File: frontend/src/types/bulk.ts
/** Bulk processing TypeScript types. */

export type BulkJobStatus = "pending" | "processing" | "completed" | "failed";

export type BulkDocumentStatus =
  | "pending"
  | "processing"
  | "completed"
  | "failed";

export interface BulkJobDocument {
  documentId: string;
  fileName: string;
  status: BulkDocumentStatus;
  errorMessage: string | null;
  processingTimeMs: number | null;
}

export interface BulkJob {
  id: string;
  status: BulkJobStatus;
  totalDocuments: number;
  processedCount: number;
  failedCount: number;
  createdAt: string;
  completedAt: string | null;
}

export interface BulkJobDetail extends BulkJob {
  documents: BulkJobDocument[];
}

export interface BulkJobListResponse {
  jobs: BulkJob[];
}

export interface BulkUploadResponse {
  jobId: string;
  status: BulkJobStatus;
  totalDocuments: number;
  documents: Array<{
    documentId: string;
    fileName: string;
    status: BulkDocumentStatus;
  }>;
  createdAt: string;
}

File: backend/src/api/schemas/bulk.py
"""Pydantic schemas for bulk processing API endpoints."""

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class BulkJobDocumentResponse(BaseModel):
    """Response schema for a single document within a bulk job."""

    document_id: uuid.UUID
    file_name: str
    status: str
    error_message: str | None = None
    processing_time_ms: int | None = None

    model_config = {"from_attributes": True}


class BulkJobResponse(BaseModel):
    """Response schema for bulk job list items."""

    id: uuid.UUID
    status: str
    total_documents: int
    processed_count: int
    failed_count: int
    created_at: datetime
    completed_at: datetime | None = None

    model_config = {"from_attributes": True}


class BulkJobDetailResponse(BaseModel):
    """Response schema for detailed bulk job view with document breakdown."""

    id: uuid.UUID
    status: str
    total_documents: int
    processed_count: int
    failed_count: int
    created_at: datetime
    completed_at: datetime | None = None
    documents: list[BulkJobDocumentResponse] = Field(default_factory=list)

    model_config = {"from_attributes": True}


class BulkUploadResponse(BaseModel):
    """Response schema for bulk upload endpoint."""

    job_id: uuid.UUID
    status: str
    total_documents: int
    documents: list[BulkJobDocumentResponse] = Field(default_factory=list)
    created_at: datetime


class BulkJobListResponse(BaseModel):
    """Response schema for listing all bulk jobs."""

    jobs: list[BulkJobResponse] = Field(default_factory=list)
